### PR TITLE
Add a limit in the graph code for number of metrics

### PR DESCRIPTION
**Pull request by dgl**
* [x] I added CHANGELOG entry for this change.

## Changes

Improvements to handle large numbers of metric names in the graph UI.

This merges my upstream (but pending) changes in Prometheus Graph UI (https://github.com/prometheus/prometheus/pull/6100, discussed a bit in https://github.com/prometheus/prometheus/issues/5919 too).

## Verification

Tested query UI with 1 million metrics (via Geras) in Chrome.

### CODE REVIEW

File: pkg/ui/static/js/graph.js [hunk 1]
```diff
@@ -332,29 +344,44 @@ Prometheus.Graph.prototype.populateInsertableMetrics = function() {
   });
 };
 
+Prometheus.Graph.prototype.makeFuzzyResult = function(self, query, source) {
+  // If fuzzyResult has already been calculated for this query, don't calculate again
```
Comment (bwplotka) [inline pkg/ui/static/js/graph.js]: missing trailing period (:

Comment (dgl) [reply pkg/ui/static/js/graph.js]: Fixed.

**Comment by GiedriusS** [comment]
The code looks good, need to try out on my cluster to see if it helps - I wonder if just the parsing of the JSON won't crash my browser.

**Comment by GiedriusS** [comment]
On an environment with ~1.5 million metrics, this definitely fixes the problem (tested it out) just that there is an interesting artifact: the dropdown disappears after a small hiccup when the JSON finishes downloading and the parsing begins :smile: 

Lets see what Prometheus's maintainers will say.

**Comment by GiedriusS** [comment]
Funny how they are talking about it but still no one reviewed your PR on the Prometheus side :crying_cat_face:

**Pull request closed without merging**